### PR TITLE
Update 0.6 branch with new namespaced user info field

### PR DESCRIPTION
**Pull request by modulitos**
**What this PR does / why we need it**:

Cherry-pick changes from: https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/743/files

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # N/A

### CODE REVIEW

File: version.txt [hunk 1]
```diff
@@ -1 +1 @@
-0.6.24
```
Comment (xdu31) [inline version.txt:1]: It looks we have the convention to cut release in a separate PR, not sure if it can work if we do it together with other changes

Comment (modulitos) [reply version.txt]: Thank for the heads up - I removed the change from the PR.

Once this PR is merged, I'll send a separate PR to bump the version again.

Comment (modulitos) [reply version.txt]: PR to only bump the version: https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/748

**Pull request merged**